mypy/types.py

```diff
@@ -1162,23 +1162,45 @@ class TypeType(Type):
     # a generic class instance, a union, Any, a type variable...
     item = None  # type: Type
 
+    # this class should not be created directly
+    # use make method which may return either TypeType or UnionType
+    # this is to ensure Type[Union[A, B]] is always represented as Union[Type[A], Type[B]]
     def __init__(self, item: Type, *, line: int = -1, column: int = -1) -> None:
+        raise NotImplementedError
+
+    def _init(self, item: Type, *, line: int = -1, column: int = -1) -> None:
         super().__init__(line, column)
         if isinstance(item, CallableType) and item.is_type_obj():
             self.item = item.fallback
         else:
             self.item = item
 
+    def __new__(cls, *args, **kwargs):  # type: ignore
+        instance = object.__new__(cls)
+        instance._init(*args, **kwargs)
+        return instance
+
+    def __copy__(self) -> 'Type':
```

Comment:
That was needed because `types.copy_type` uses `copy.copy(t)`, and the default `copy` simply returns the (unchanged) original object which is not the right semantics here (it breaks join_types).